src/main/MQ2ImGuiConsole.cpp

```diff
@@ -27,1504 +27,1638 @@
+							WriteChatColor("\awWHITE     \a-wGREY");
+
+							MakeColorGradient(.3f, .3f, .3f, 0, 2, 4);
 						}
-						if (ImGui::MenuItem("Achievement link Test"))
+
+						if (m_zepEditor)
 						{
-							m_zepEditor->DoAchievementLinkTest();
+							if (ImGui::MenuItem("Hyperlink Test"))
+							{
+								m_zepEditor->DoHyperlinkTest();
+							}
+							if (ImGui::MenuItem("Achievement link Test"))
+							{
+								m_zepEditor->DoAchievementLinkTest();
+							}
 						}
-					}
 
+						ImGui::EndMenu();
+					}
 					ImGui::EndMenu();
 				}
-				ImGui::EndMenu();
-			}
-
-			if (ImGui::BeginMenu("Windows"))
-			{
-				ImGuiTools_DrawWindowsMenu();
-				ImGui::EndMenu();
-			}
-
-			DeveloperTools_DrawMenu();
-
-			ImGui::EndMenuBar();
-		}
 
-		// And put it back ...
-
-		const float footer_height_to_reserve = ImGui::GetStyle().ItemSpacing.y + ImGui::GetFrameHeightWithSpacing(); // 1 separator, 1 input text
-
-		// Right click menu for editor
-		if (ImGui::BeginPopupContextWindow())
-		{
-			if (ImGui::Selectable("Clear")) ClearLog();
-			ImGui::EndPopup();
-		}
-
-		ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
+				if (ImGui::BeginMenu("Windows"))
+				{
+					ImGuiTools_DrawWindowsMenu();
+					ImGui::EndMenu();
+				}
 
-		ImVec2 contentSize = ImGui::GetContentRegionAvail();
-		contentSize.y -= footer_height_to_reserve;
+				DeveloperTools_DrawMenu();
 
-		m_zepEditor->Render("##ZepConsole", contentSize);
+				ImGui::EndMenuBar();
+			}
 
-		// Command-line
-		ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(2, 4));
-		ImGui::Separator();
+			// And put it back ...
 
-		int textFlags = ImGuiInputTextFlags_EnterReturnsTrue
-			| ImGuiInputTextFlags_CallbackCompletion
-			| ImGuiInputTextFlags_CallbackHistory;
+			const float footer_height_to_reserve = ImGui::GetStyle().ItemSpacing.y + ImGui::GetFrameHeightWithSpacing(); // 1 separator, 1 input text
 
-		ImGui::SetCursorPosX(ImGui::GetCursorPosX() + 6);
-		ImGui::SetCursorPosY(ImGui::GetCursorPosY() + 4);
-		ImGui::PushItemWidth(ImGui::GetContentRegionAvailWidth());
-		ImGui::PushStyleColor(ImGuiCol_FrameBg, ImVec4(0.00f, 0.00f, 0.00f, 0.00f));
-		ImGui::PushFont(mq::imgui::ConsoleFont);
+			// Right click menu for editor
+			if (ImGui::BeginPopupContextWindow())
+			{
+				if (ImGui::Selectable("Clear")) ClearLog();
+				ImGui::EndPopup();
+			}
 
-		bool bTextEdit = ImGui::InputText("##Input", m_inputBuffer, IM_ARRAYSIZE(m_inputBuffer), textFlags,
-			[](ImGuiInputTextCallbackData* data)
-		{ return static_cast<ImGuiConsole*>(data->UserData)->TextEditCallback(data); }, this);
+			ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
 
-		ImGui::PopFont();
-		ImGui::PopStyleColor();
-		ImGui::PopStyleVar(2);
+			ImVec2 contentSize = ImGui::GetContentRegionAvail();
+			contentSize.y -= footer_height_to_reserve;
 
+			// Set Font for ZepEditor
+			m_zepEditor->mq::imgui::ImGuiZepEditor::SetFont(Zep::ZepTextType::Text, ImGui::GetFont(), s_consoleFontSize);
```

Comment:
SetFont should only be called if the font is being changed.

`m_zepEditor->SetFont`, the whole function scope doesn't need to be qualified.